apps/sms/views/conversation/js/conversation.js

```diff
@@ -2014,13 +2036,25 @@ var ConversationView = {
     ).then(() => {
       WaitingScreen.show();
 
-      var messageIdsToDelete = this.selectionHandler.selectedList.map(
-        (item) => +item
-      );
-
-      return this.deleteMessages(messageIdsToDelete);
+      if(ConversationView.inEditMode) {
+        messageIdsToDelete = this.selectionHandler.selectedList.map(
+          (item) => +item
+        );
+        return onAllThreadMessagesRetrieved(messageIdsToDelete);
+      } else {
+        console.log('test1');
+        MessageManager.getMessages({
+        // Filter and request all messages with this threadId
+          filter: { threadId: ConversationView.activeThread.id },
+          each: onThreadMessageRetrieved,
+          end: onAllThreadMessagesRetrieved
```

Comment:
> what do you mean by deleteConversation method exactly. Do we have any method like this already?

No we don't, we have very similar one in inbox, so we should have very similar one in Conversation view as well.

>  or you mean to create a new method, if yes then what it will do differently. 

Yes, explained above - see the one in Inbox view - and just take what you need from it - I guess almost only retrieve all message Ids and call ```MessageManager.deleteMessages(messageIdsToDelete);``` - but you need to check and find out for sure :)

>  And what useless work 'deleteMessages' does?

Just take a look at ```ConversationView.deleteMessages``` and you'll see :) When you remove entire thread you __don't need__ to:
* Remove Dom node for every message - just show wait screen and then go back to inbox when deletion is finished;
* Check message list length to decide whether or not you should go Inbox - you know for sure!;
* Do very expensive ```lastDateGroup.querySelector('li:last-child')``` ;
* Do ```InboxView.updateThread``` - it will be removed anyway :)

So as you see almost 99% of useless work for the delete-entire-thread case :)

And when you see three ```if(ConversationView.inEditMode)``` in a single method - it's a strong sign that method should be split - the main rule here is that common staff stays in a single place, very specific staff is moved to separate methods - here you just should use your own judgment on whether it looks good or not - good exercise :)